GetOldTweets3/manager/ConcurrentTweetManager.py

```diff
@@ -0,0 +1,80 @@
+# Author: Giulio Neusch-Frediani - www.github.com/giulionf
+from datetime import datetime
+from threading import Thread
+from queue import Queue
+from queue import Empty
+import copy
+
+from GetOldTweets3.manager.TweetManager import TweetManager
+
+DATE_FORMAT = "%Y-%m-%d"
+FIRST_TWEET_DATE = datetime.strptime("2006-03-21", DATE_FORMAT)
+
+time_spans = Queue()
+tweets = []
+workers = []
+
+
+class ConcurrentTweetManager:
+
+    @staticmethod
+    def get_tweets(tweet_criteria, receive_buffer=None, buffer_length=100, proxy=None, debug=False, worker_count=1):
+        # Split the date in smaller parts
+        since_date = datetime.strptime(tweet_criteria.since, DATE_FORMAT) if tweet_criteria.since else FIRST_TWEET_DATE
+        until_date = datetime.strptime(tweet_criteria.until, DATE_FORMAT) if tweet_criteria.until else datetime.now()
+        date_diff_per_worker = (until_date - since_date) / worker_count
+
+        # Create a TweetCriteria that can be cloned as Model by the Workers
+        criteria = copy.deepcopy(tweet_criteria)
+        criteria.set_max_tweets(criteria.max_tweets / worker_count)
+
+        # Create a time span for each of the splitted parts and a corresponding worker
+        for i in range(1, worker_count+1):
+            global time_spans
+            global workers
+            from_time = since_date + (i-1) * date_diff_per_worker
+            to_time = since_date + i * date_diff_per_worker
+            time_spans.put((from_time, to_time))
+
+            w = WorkerThread(copy.deepcopy(criteria), receive_buffer, buffer_length, proxy, debug)
+            w.start()
+            workers.append(w)
+
+        # Wait for the workers to finish and then stop them
+        time_spans.join()
+        for worker in workers:
+            worker.stop()
+
+        return tweets
```

Comment:
You're right! I will implement that asap!

Reply:
... and I don't very like the idea of storing tweets in memory during retrieval (which could possibly last for a day). If something goes wrong all the retrieved tweets will lost. The current implementation with saving tweets by 100 in a batch is pretty robust.

Reply:
Isn't this done by the TweetManagers though? I guessed since the ConcurrentTweetManager just splits the request into parts and handles it with multiple sub-TweetManagers they all have the batches as they used to... The user should still be able to use the receiveBuffer as he can with the Non concurrent Tweetmanager... Or am I missing something?

Reply:
We have a call to `receiveBuffer` in `getTweets()` for saves.
```python
if receiveBuffer and len(resultsAux) > 0:
    receiveBuffer(resultsAux)
    resultsAux = []
```

The return value from getTweets() is not used https://github.com/Mottl/GetOldTweets3/blob/c9b829506792c402ae6661a8c7ea228f57c43f0b/bin/GetOldTweets3#L170-L171